### PR TITLE
SSCS-10057-Implement-Hearing-GET-Endpoint-and-Contract-Tests-master-rebased-20220321

### PR DESCRIPTION
**Pull request by bensonlievoco**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10057

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/HearingResponse.java [hunk 1]
```diff
@@ -21,4 +24,22 @@
...
     private Number versionNumber;
+
+    private String listAssistTransactionID;
+
+    private LocalDateTime receivedDateTime;
+
+    private Integer responseVersion;
+
+    @JsonProperty("laCaseStatus")
+    @EnumPattern(enumClass = ListingCaseStatus.class, fieldName = "listingCaseStatus")
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/HearingResponse.java:35]: Should these two be enums?

Comment (bensonlievoco) [reply src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/HearingResponse.java:35]: in my mapping, they are enum
image

File: src/contractTest/java/uk/gov/hmcts/reform/sscs/consumer/GetHearingPactConsumerTest.java [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+        HearingGetResponse expected = objectMapper.readValue(
+            generateValidHearingGetResponsePactDslJsonBody(date).toString(),
+            HearingGetResponse.class
+        );
+
+        Assertions.assertEquals(expected, result);
```
Comment (wmarshid) [inline src/contractTest/java/uk/gov/hmcts/reform/sscs/consumer/GetHearingPactConsumerTest.java:93]: Very Minor: static import assertions and remove new lines underneath please - same applies to other tests

**Review by wmarshid — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/HearingResponse.java (hunk 1 above)
Comment (dattipoeHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/HearingResponse.java:35]: I might be missing something but I was asking why they are String types when they seem to be pointing at an Enum type? Shouldn't it be private ListingCaseStatus listingCaseStatus?

Comment (bensonlievoco) [reply src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/HearingResponse.java:35]: I see other enum in the repo are use the same way, so I just follow. 
I think without using a enum will allow a bit of flexible that the data other than the enum?

**Review by dattipoeHMCTS — DISMISSED** [review]

**Review by wmarshid — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java [hunk 3]
```diff
@@ -35,4 +37,12 @@ HearingGetResponse getHearingRequest(
         @RequestParam("id") String id
     );
 
+    @DeleteMapping(value = "/hearing", consumes = MediaType.APPLICATION_JSON_VALUE)
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java:40]: Is this change meant to be included in this PR?

**Review by wmarshid — APPROVED** [review]

**Review by gokul-sol — COMMENTED** [review]
@bensonlievoco Which pact provider does these tests verify against and also are these contract tests publish to pact broker ?

File: src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java (hunk 3 above)
Comment (bensonlievoco) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/HmcHearingApi.java:40]: it just update from master, nothing I changed

**Review by dattipoeHMCTS — DISMISSED** [review]

**Review by wmarshid — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/HearingDaySchedule.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.NoArgsConstructor;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@NoArgsConstructor
+public class HearingDaySchedule {
+    private LocalDateTime hearingStartDateTime;
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/model/single/hearing/HearingDaySchedule.java:14]: Minor: new line missing between these fields

**Review by wmarshid — APPROVED** [review]

**Review by dattipoeHMCTS — APPROVED** [review]

**Review by dattipoeHMCTS — APPROVED** [review]

**Review by wmarshid — APPROVED** [review]

**Pull request merged**